### PR TITLE
[Distributed] Test if xm.all_reduce produces a single dynamo graph

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
Test if xm.all_reduce produces a single dynamo graph.

Test Plan:
PJRT_DEVICE=TPU python test/test_mp_replication.py

### CODE REVIEW

File: test/test_mp_replication.py [hunk 1]
```diff
@@ -13,23 +17,32 @@ def _mp_fn(index):
...
     xtwos = twos.to(device)
     xthrees = threes.to(device)
     xfours = fours.to(device)
+    xfives = fives.to(device)
     xm.all_reduce(xm.REDUCE_SUM, [xones, xtwos])
-    xthrees = xm.all_reduce(xm.REDUCE_SUM, xthrees)
+    xthrees = all_reduce(xthrees)
     xfours = xm.all_reduce(xm.REDUCE_SUM, xfours, scale=scale)
 
+    compiled_all_reduce = torch.compile(
```
Comment (JackCaoG) [inline test/test_mp_replication.py:31]: I would prefer move dynamo tests under `test/dynamo`, we can do that in a follow up pr.

**Review by JackCaoG — APPROVED** [review]

File: test/test_mp_replication.py (hunk 1 above)
Comment (alanwaketan) [reply test/test_mp_replication.py:31]: Sure, will do that as a follow up.

**Comment by alanwaketan** [comment]
Thanks Jack for approving the change.

**Comment by alanwaketan** [comment]
The tests are all pass in the CI now. I will just land it directly after rebasing.

**Pull request merged**